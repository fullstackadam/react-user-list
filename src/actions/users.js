import uuid from 'uuid';
import moment from 'moment';

export const addUser = (
  { 
    firstName,
    lastName = '',
    address = '', 
    city, 
    state,
    zip = '', 
  } = {}
) => ({
  type: 'ADD_USER',
  user: {
    id: uuid(),
    firstName,
    lastName,
    address,
    city,
    state,
    zip,
    createdAt: moment()
  }
});

export const removeUser = ({ id } = {}) => ({
  type: 'REMOVE_USER',
  id
});

export const editUser = (id, updates) => ({
  type: 'EDIT_USER',
  id,
  updates
});
