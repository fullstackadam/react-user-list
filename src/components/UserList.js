import React from 'react';
import { connect } from 'react-redux';
import selectUsers from '../selectors/users';
import UserListItem from './UserListItem';

const UserList = (props) => (
  <div>
    {props.users.map((user) => <div key={user.id}><UserListItem {...user} /><hr /></div>)}
  </div>
);

const mapStateToProps = (state) => {
  return {
    users: selectUsers(state.users, state.filters)
  };
};

export default connect(mapStateToProps)(UserList);