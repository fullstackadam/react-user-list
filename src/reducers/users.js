const usersReducerDefaultState = [];

export default (state = usersReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_USER':
      return state.concat(action.user);
      break;
    case 'REMOVE_USER':
      return state.filter(({ id }) =>  id !== action.id);
      break;
    case 'EDIT_USER':
      return state.map((user) => {
        if (user.id === action.id) {
          return {
            ...user,
            ...action.updates
          };
        } else {
          return user;
        }
      });
    default:
      return state;
  }
};