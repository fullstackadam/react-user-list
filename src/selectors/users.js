export default (users, { text }) => {
  return users.filter((user) => {
    const textMatch = user.firstName.toLowerCase().includes(text.toLowerCase())
                      || user.lastName.toLowerCase().includes(text.toLowerCase());

    return textMatch;
  });
};