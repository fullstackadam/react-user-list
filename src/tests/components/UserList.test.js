import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from '../../store/configure';
import { addUser } from '../../actions/users';
import UserList from '../../components/UserList';
import users from '../fixtures/users';

test('should render UserList with users', () => {
  const store = configureStore();

  store.dispatch(addUser(users[0]));
  store.dispatch(addUser(users[1]));
  store.dispatch(addUser(users[2]));

  const wrapper = shallow(
    <Provider store={store}>
      <UserList />
    </Provider>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});