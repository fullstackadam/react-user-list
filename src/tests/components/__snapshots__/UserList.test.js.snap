// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserList with users 1`] = `<Connect(UserList) />`;
